libase/tds/field.go

```diff
@@ -709,8 +709,38 @@ type VarBinaryFieldData struct{ fieldData }
 type VarCharFieldData struct{ fieldData }
 type BigDateTimeNFieldData struct{ fieldData }
 type BigTimeNFieldData struct{ fieldData }
-type DecNFieldData struct{ fieldData }
-type NumNFieldData struct{ fieldData }
+
+type fieldDataPrecisionScale struct {
+	fieldData
+}
+
+func (field *fieldDataPrecisionScale) ReadFrom(ch BytesChannel) (int, error) {
+	n, err := field.readFrom(ch)
+	if err != nil {
+		return 0, err
```

Comment:
`return n, err`.

If `err` is `io.EOF` the accumulated number of read bytes is used to print an error - which would be incorrect if `n` isn't returned.